phoenix-pherf/src/main/java/org/apache/phoenix/pherf/util/ResultsComparator.java

```diff
@@ -0,0 +1,353 @@
+		String footer = StaticGoogleChartsRenderingData.FOOTER
+				.replace("[summary]",
+						((verifyWithinThreshold(threshold) == true ? "<font color=green>PASSED | Results are within ": 
+							"<font color=red>FAILED | Results are outside "))
+								+ "set threshold of " + thresholdString + "</font><br>"
+								+ new SimpleDateFormat("yyyy/MM/dd ha z").format(new Date()));
+		footer = footer.replace("[title]", labels[0]);
+    	writer.println(footer);
+    	writer.close();
+    }
+    
+    /**
+     * Render a blank Google charts row
+     * @return
+     */
+    private String getBlankRow() {
+    	String ret = "['" + new String(new char[60]).replace("\0", ".") + "'";
+    	for (int i=0; i<labels.length; i++)
+    		ret += ",0,''";
+    	ret += "],";
+    	return ret;
+	}
+
+    /**
+     * Render tooltip as HTML table
+     * @param nodeDataSet
+     * @return
+     */
+	private String getToolTipAsHTML(Map<String, Node> nodeDataSet) {
+       	String ret = "<table width=1000 cellpadding=1 cellspacing=0 border=0 bgcolor=#F4F4F4><tr>";
+    	for (Map.Entry<String, Node> nodeSet : nodeDataSet.entrySet())	
+    		ret += "<td>" + getToolText(nodeSet.getValue()) + "</td>";
+    	return ret + "</tr></table>";
+    }
+    
+	/**
+	 * Get tooltip for node
+	 * @param node
+	 * @return
+	 */
+    private String getToolText(Node node) {
+        return node.getLabelAsHTML() 
+        		+ node.getAvgTimeAsHTML()
+        		+ node.getMinTimeAsHTML()
+        		+ node.getNumRunsAsHTML()
+        		+ node.getRowCountAsHTML()
+        		+ node.getExplainPlanAsHTML()
+        		+ node.getQueryAsHTML();
+    }
+    
+    /**
+     * DataNode to store results to render and compare 
+     */
+    class DataNode {
+    	private Map<String, Node> dataSet = new LinkedHashMap<String, Node>();
+		
+    	public DataNode(String label, Node node) {
+    		this.getDataSet().put(label, node);
+    	}
+    	
+		public Map<String, Node> getDataSet() {
+			return dataSet;
+		}
+		public void setDataSet(Map<String, Node> dataSet) {
+			this.dataSet = dataSet;
+		}
+    }
+    
+    class Node {
+    	private String explainPlan;
+    	private String query;
+    	private String tenantId;
+    	private long minTime;
+    	private long avgTime;
+    	private long numRuns;
+    	private long rowCount;
+    	private String label;
+    	private DecimalFormat df = new DecimalFormat("#.#");
+    	
+    	public Node(long minTime, long avgTime, long numRuns, String explainPlan, String query, String tenantId, String label, long rowCount) {
+    		this.setMinTime(minTime);
+    		this.setAvgTime(avgTime);
+    		this.setNumRuns(numRuns);
+    		this.setExplainPlan(explainPlan);
+    		this.setQuery(query);
+    		this.setTenantId(tenantId);
+    		this.setLabel(label);
+    		this.setRowCount(rowCount);
+    	}
+    	
+		String getExplainPlan() {
+			return explainPlan;
+		}
+		String getExplainPlanAsHTML() {
+			return "</br><font face=arial size=1><b>EXPLAIN PLAN </b>"
```

Comment:
Perhaps we should use an html generation library instead of hard coding all this in strings?

Never tried this one, bit looks like it has god template support.
http://freemarker.org/docs/dgui_quickstart_basics.html